File: trikot-viewmodels-declarative-flow/viewmodels-declarative-flow/src/tvosMain/kotlin/com/mirego/trikot/viewmodels/declarative/configuration/VMDImageProvider.kt
package com.mirego.trikot.viewmodels.declarative.configuration

import com.mirego.trikot.viewmodels.declarative.properties.VMDImageResource
import platform.UIKit.UIImage

actual interface VMDImageProvider {
    fun imageForResource(imageResource: VMDImageResource): UIImage?
}

File: trikot-viewmodels-declarative-flow/swift/swiftui/ImageResource/VMDImageProvider.swift
import SwiftUI
import TrikotFrameworkName

public protocol VMDImageProvider {
    func imageForResource(imageResource: VMDImageResource) -> Image?
}

public struct DefaultImageProvider: VMDImageProvider {
    public init() { }
    public func imageForResource(imageResource: VMDImageResource) -> Image? { nil }
}
